src/subscriptions.ts

```diff
@@ -185,7 +185,7 @@ export async function getSubscribedTransactions(
   if (!skipAlgodSync) {
     start = +new Date()
     const blocks = await getBlocksBulk({ startRound: algodSyncFromRoundNumber, maxRound: endRound }, algod)
-    const blockTransactions = blocks.flatMap((b) => getBlockTransactions(b))
+    const blockTransactions = blocks.flatMap((b) => [getTransactionFromBlockPayout(b), ...getBlockTransactions(b)])
```

Comment:
```suggestion
    const blockTransactions = blocks.flatMap((b) => getBlockTransactions(b).concat(getTransactionFromBlockPayout(b)))
```

Reply:
Do all blocks have a proposer payout pay transaction? Based on my observations it appears indexer only add this transaction when the proposer is rewards elgible.

Reply:
Good catch on both. I realized the latter when working on the python implementation and checked if the payout was zero or not (it gets zeroed out when proposer is not eligible). Will do the same here

Reply:
Both addressed in [6810844](https://github.com/algorandfoundation/algokit-subscriber-ts/pull/110/commits/6810844238e22dbbc6ecd5d84d87e559269cde58). Also added a check to the round offset in the testcase

Reply:
Looks great.
I also just installed into my local lora and double checked everything works correctly.